### PR TITLE
Update site mobile menu

### LINKED ISSUE #1183: Implement mobile navigation menu
This is the implementation story for the **mobile version** of the nav updates: https://github.com/Qiskit/qiskit.org/issues/862

Acceptance criteria
- [ ] Navigation looks like the Figma file.
- [ ] Navigation displays Events and Advocates links.

Figma file: https://www.figma.com/file/PDbN8wcg5WteWVvinjAEUv/qiskit.org-main--file?node-id=1%3A9533

### LINKED ISSUE #1240: Implement open states for mobile dropdown component
Related to #1183 

This story is to handle the open states for the mobile menu dropdown component

- [ ] the Community dropdown should be open by default, revealing both sublinks
- [ ] clicking on Community should toggle the dropdown component

### PR DESCRIPTION
**Pull request by techtolentino**
Fixes #1183 and fixes #1240

Notes: 
- this PR will only update the mobile nav by updating its position, along with adding the `cv-dropdown` component and overriding styles
- There's also handles the toggling states of the mobile dropdown component

### CODE REVIEW

**Comment by techtolentino** [comment]
@JRussellHuffman - tagging you here to let you know of an issue I ran into (please see notes above, 2nd bullet point).

You can also view the mobile nav updates [here](https://qiskitorg-git-rt-1183-mobile-nav-updates.qiskit.now.sh/)

**Review by y4izus — CHANGES_REQUESTED** [review]
Thanks Randy!

Could you check with Russell if the menu is ok for mobile devices? On Figme he has this design:

Screen Shot 2020-12-10 at 11 55 42

And we have the same menu that we have for medium size:

Screen Shot 2020-12-10 at 11 56 06

File: components/layouts/TheMenu/MobileMenu.vue [hunk 1]
```diff
@@ -5,52 +5,83 @@
...
+export default class MobileMenu extends Mixins(MenuMixin) {
+  @Prop({ type: Boolean, default: false })
+  isMobileMenuVisible: boolean=false
+
+  stayConnectedElements = STAY_CONNECTED_LINKS
+  theme = 'light'
+
+  @Watch('isMobileMenuVisible')
+  showDropdown () {
+  /**
```
Comment (delapuente) [inline components/layouts/TheMenu/MobileMenu.vue:74]: Avoid explaining what the function does in a comment, change the name of the function to something like `openDropdown` and you will remove the need for further explanations and, by the way, prevent the name collision with the const `showDropMenu`.

File: components/layouts/TheMenu/MobileMenu.vue [hunk 2]
```diff
@@ -5,52 +5,83 @@
...
 
+import {
+  STAY_CONNECTED_LINKS
+} from '~/constants/menuLinks'
+
 @Component
-export default class MobileMenu extends Mixins(MenuMixin) {}
+export default class MobileMenu extends Mixins(MenuMixin) {
+  @Prop({ type: Boolean, default: false })
+  isMobileMenuVisible: boolean=false
```
Comment (delapuente) [inline components/layouts/TheMenu/MobileMenu.vue:67]: The component is the `MobileMenu` already so you can remove context from the name and simply call it `isVisible`.

File: components/layouts/TheMenu/MobileMenu.vue [hunk 3]
```diff
@@ -5,52 +5,83 @@
...
 import MenuMixin from '~/mixins/menu'
 
+import {
+  STAY_CONNECTED_LINKS
+} from '~/constants/menuLinks'
+
 @Component
-export default class MobileMenu extends Mixins(MenuMixin) {}
+export default class MobileMenu extends Mixins(MenuMixin) {
+  @Prop({ type: Boolean, default: false })
```
Comment (delapuente) [inline components/layouts/TheMenu/MobileMenu.vue:66]: Usually, we write the property name following the decorator. Take a look at other components to see how.

File: components/layouts/TheMenu/index.vue [hunk 4]
```diff
@@ -98,6 +99,7 @@ import MenuMixin from '~/mixins/menu'
 @Component
 export default class TheMenu extends Mixins(MenuMixin) {
   isMobileMenuVisible: boolean = false
+  isCommunityDropdownVisible = false
```
Comment (delapuente) [inline components/layouts/TheMenu/index.vue:102]: I did not find any use of this field in the code. Is it safe to remove completely?

**Review by delapuente — CHANGES_REQUESTED** [review]
Nice work, @techtolentino. You did it! Please, address my comments below.

**Review by delapuente — APPROVED** [review]
Nice done @techtolentino. Happy to see the mobile menu in place!

**Comment by delapuente** [comment]
@y4izus would you mind double-checking the visual implementation?

**Comment by y4izus** [comment]
@techtolentino , could you double-check with Russell this styles:

- On medium size: Check if the bottom content is correct
  - Figma design
  
Screen Shot 2021-01-25 at 09 56 43

  - PR design
Screen Shot 2021-01-25 at 09 57 10

- On small size: Check if the footer should be sticky to the bottom
  - Figma design
  
Screen Shot 2021-01-25 at 09 58 32

  - PR design
  
Screen Shot 2021-01-25 at 09 59 23

**Review by y4izus — APPROVED** [review]
Thank you Randy!

**Pull request merged**